Goutte/Client.php

```diff
@@ -62,6 +62,19 @@ public function removeHeader($name)
         unset($this->headers[$name]);
     }
 
+    public function resetHeaders()
+    {
+        $this->headers = array();
+
+        return $this;
+    }
+
+    public function reset()
+    {
```

Comment:
you also need to call the parent reset method, otherwise cookies are not reset properly